### DISCUSSION TITLE
Searching files without UTF-8 BOM?

### DISCUSSION THREAD

**Discussion by Liturgist** [community/community#51859] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

Documentation says that only UTF-8 encoded files are searched.

Does that mean that a UTF-8 BOM (Byte Order Mark) is required at the beginning of the file? It does not appear so.

Is a file omitted from search if a non-UTF-8 character sequence is found?

**Comment by look** [comment, marked as answer]
> Is a file omitted from search if a non-UTF-8 character sequence is found?

That's correct. We convert the raw bytes to UTF-8 and if the conversion is not successful, the blob is excluded.

**Reply by Liturgist** [reply]
This seems to suggest that files without a Unicode BOM are searched. Is that correct?

How is it determined what the incoming encoding is? Converting from ??? to UTF-8.

**Reply by look** [reply]
The files are validated with Go's [`utf8.Valid()`](https://pkg.go.dev/unicode/utf8#Valid). This checks the byte sequence to ensure it is valid UTF-8. Non UTF-8 encodings are not transcoded. I validated that this function does work with a byte order mark, though these are [generally not recommended](https://stackoverflow.com/questions/2223882/whats-the-difference-between-utf-8-and-utf-8-with-bom) for UTF-8 encoded files.
